components/navigation/NavbarLinks.tsx

```diff
@@ -13,7 +13,7 @@ const NavbarLinks = ({
       className={`mx-3.5 ${
         showMobileNavbar ? 'mt-6' : ''
       } flex h-full items-center text-[0.875rem] font-normal text-white-500 dark:text-white-800 ${
-        isActive && 'font-semibold text-primary-light dark:text-primary-dark'
+        isActive && 'font-semibold text-[#0252CD] dark:text-[#428DFF]'
```

Comment:
To answer your question- this is because your `isActive` state isn't **toggling** these classes, you're simply adding them to the mix.

So your links have both classes telling to to have a white text, and a blue text. The browser can't ascertain the precedence you want the active state to have. 

You can either:

1) Create slightly more complex logic - use a ternary.
```js
   <p
      className={`mx-3.5 ${
        showMobileNavbar ? 'mt-6' : ''
      } flex h-full items-center text-[0.875rem] ${
        isActive
          ? 'font-semibold text-[#0252CD] dark:text-[#428DFF]'
          : ' font-normal text-white-500 dark:text-white-800'
      }`}
    >
      {label}
    </p>
```

2) Use the `bang` character to tell the browser this section is `!important`:
```js
  <p
      className={`mx-3.5 ${
        showMobileNavbar ? 'mt-6' : ''
      } flex h-full items-center text-[0.875rem] font-normal text-white-500 dark:text-white-800 ${
        isActive && 'font-semibold !text-[#0252CD] dark:!text-[#428DFF]'
      }`}
    >
      {label}
    </p>
```

Reply:
Rule 'o Thumb - never use ! - true?

First solution is more readable - clearer